src/docbuilder/rustwide_builder.rs

```diff
@@ -343,26 +343,23 @@ impl RustwideBuilder {
             .run(|build| {
                 use crate::docbuilder::metadata::BuildTargets;
 
-                let mut files_list = None;
                 let mut has_docs = false;
-                let mut algs = CompressionAlgorithms::default();
                 let mut successful_targets = Vec::new();
                 let metadata = Metadata::from_source_dir(&build.host_source_dir())?;
                 let BuildTargets {
                     default_target,
                     other_targets,
                 } = metadata.targets();
 
-                // Do an initial build and then copy the sources in the database
+                // Store the sources even if the build fails
+                debug!("adding sources into database");
+                let prefix = format!("sources/{}/{}", name, version);
+                let (files_list, mut algs) =
+                    add_path_into_database(&self.storage, &prefix, build.host_source_dir())?;
```

Comment:
It feels like this should still be done down around `add_package_into_database` until we proactively insert the release details into the database before building, otherwise in cases where we completely fail and don't insert the release details there will be uploaded files unassociated with any release.

Reply:
Hmm, that seems like a different refactor? The error case you're talking about makes sense, but if I move uploading the sources later we'll still upload all the documentation if an error occurs.

Reply:
And actually I don't see a way to have `add_package` come first because we record whether or not the build succeeded. So many we should the steps of building the docs and uploading the docs? Have one come before update the DB and the other come after? That definitely seems like a larger PR though.

Reply:
It seems like the `execute_build` call is the most likely to fail, so specifically doing the upload before there has the highest chance of getting uploaded files unassociated with a release.

Reply:
Updated!